kerastuner/tuners/bayesian.py

```diff
@@ -0,0 +1,289 @@
+            values = {}
+            for p in self.space:
+                values[p.name] = p.random_sample(self._seed_state)
+                self._seed_state += 1
+            # Keep trying until the set of values is unique,
+            # or until we exit due to too many collisions.
+            values_hash = self._compute_values_hash(values)
+            if values_hash in self._tried_so_far:
+                collisions += 1
+                if collisions > self._max_collisions:
+                    return None
+                continue
+            self._tried_so_far.add(values_hash)
+            break
+        return values
+
+    def _get_training_data(self):
+        x = []
+        y = []
+        for trial_id in self._values:
+            values = self._values[trial_id]
+            if trial_id not in self._score:
+                continue
+            score = self._score[trial_id]
+
+            vector = [0] * len(self.space)
+            for name, value in values.items():
+                index = self._get_hp_index(name)
+                hp = self.space[index]
+                if isinstance(hp, hp_module.Choice):
+                    value = hp.values.index(value)
+                elif isinstance(hp, hp_module.Fixed):
+                    value = 0
+                elif isinstance(hp, hp_module.Range):
+                    value = value
+                elif isinstance(hp, hp_module.Linear):
+                    value = value
+                vector[index] = value
+
+            # Exclude the fixed value from the vector
+            no_fixed_vector = []
+            for index, value in enumerate(vector):
+                if not isinstance(self.space[index], hp_module.Fixed):
+                    no_fixed_vector.append(value)
+            x.append(no_fixed_vector)
+            y.append(score)
+        x = np.array(x)
+        y = np.array(y) / np.max(y)
+        return x, y
+
+    def _convert(self, vector):
+        values = {}
+        vector_index = 0
+        for index, hp in enumerate(self.space):
+            hp = self.space[index]
+            if isinstance(hp, hp_module.Fixed):
+                values[hp.name] = hp.value
+                continue
+
+            value = vector[vector_index]
+            vector_index += 1
+
+            if isinstance(hp, hp_module.Choice):
+                value = hp.values[int(value)]
+            elif isinstance(hp, hp_module.Linear):
+                value = value[0]
+            elif isinstance(hp, hp_module.Range):
+                value = value[0]
+            values[hp.name] = value
+
+        return values
+
+    def _get_hp_index(self, name):
+        for index, hp in enumerate(self.space):
+            if hp.name == name:
+                return index
+        return None
+
+    def _upper_confidence_bound(self, x):
+        x = x.reshape(-1, self._dim)
+        try:
+            mu, sigma = self.gpr.predict(x, return_std=True)
+        except UserWarning:
```

Comment:
Warnings are not exceptions (at least not by default). Is this tested?

Perhaps there is a simpler way to detect when default values should be used? E.g. if no training data is available.

Reply:
It was caused by large values of y and it is fixed by normalizing y. I will remove the except.